### PR TITLE
Feature 456 : Add labels from label-schema.org to the generator images

### LINKED ISSUE #456: Add labels from label-schema.org to the generators
microbadger looks cool 
https://microbadger.com/

and has a set of [canonical docker labels](https://microbadger.com/labels) we should automatically add into our docker images as part of a build

### PR DESCRIPTION
**Pull request by dev-gaur**
Fixes #456

### CODE REVIEW

**Comment by dev-gaur** [comment]
image

Its still a WIP. The labels aren't coming up as expected. There's a twist :stuck_out_tongue:

File: generator/java-exec/src/main/java/io/fabric8/maven/generator/javaexec/JavaExecGenerator.java [hunk 1]
```diff
@@ -16,10 +16,7 @@
 package io.fabric8.maven.generator.javaexec;
 
 import java.io.File;
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
```
Comment (rohanKanojia) [inline generator/java-exec/src/main/java/io/fabric8/maven/generator/javaexec/JavaExecGenerator.java]: What's the purpose of this change? I usually avoid using wildcard imports since they clutter your namespace.

Comment (dev-gaur) [reply generator/java-exec/src/main/java/io/fabric8/maven/generator/javaexec/JavaExecGenerator.java]: My IDE must have done it and I missed to check that. Fixing right away.

File: core/src/main/java/io/fabric8/maven/core/util/BuildLabelUtil.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        }
+        if (project.getOrganization() != null && project.getOrganization().getName() != null) {
+            labels.put(BuildLabelAnnotations.VENDOR.value(), project.getOrganization().getName());
+        }
+        labels.put(BuildLabelAnnotations.VERSION.value(), project.getVersion());
+        labels.put(BuildLabelAnnotations.SCHEMA_VERSION.value(), LABEL_SCHEMA_VERSION);
+
+        try {
+            String gitRemote= "origin";
+            Repository repository = GitUtil.getGitRepository(project.getBasedir());
```
Comment (rohanKanojia) [inline core/src/main/java/io/fabric8/maven/core/util/BuildLabelUtil.java]: Could you please elaborate why gitRemote is being hardcoded here? same for LABEL_SCHEMA_VERSION above. Can we put these in static constants for better access?

**Comment by rohanKanojia** [comment]
@dev-gaur : Is this still a WIP?

File: core/src/main/java/io/fabric8/maven/core/util/BuildLabelUtil.java (hunk 2 above)
Comment (dev-gaur) [reply core/src/main/java/io/fabric8/maven/core/util/BuildLabelUtil.java]: These variable are only being used one. So its better to keep them under function call stack instead of Class static memory.

**Comment by dev-gaur** [comment]
> @dev-gaur : Is this still a WIP?

Docker builds are working fine. S2I builds have some irregularities. I think I should once more if s2i errors is from our side or S2I's.

**Comment by dev-gaur** [comment]
Up for review @rohanKanojia @lordofthejars 

This PR will add labels on, Kubernetes mode, and Openshift mode with Docker build strategy case.
